### PR TITLE
#160536352 Implement admin create meals

### PR DESCRIPTION
**Pull request by abdulfataiaka**
**What does the PR do?**
- Add functionality to enable `admin` add meals

**Description of Task to be completed?**
- Update add meal component to connect to store
- Define action creators to update the store
- Make a post request to add a meal
- Handle response from post request

**How should this be manually tested?**
- Clone the repository [here](https://github.com/andela/AndelaEatsUI.git)
- Change directory into AndelaEatsUI and checkout to this branch `ft-admin-create-meal-160536352`
- Run `npm run bundle` and `npm run dev`
- Log in as an `Admin`, navigate to `admin dashboard`, select `Meals`, and click on `Add meal`
- Complete the form and click the `Add meal` button
- run `npm test` to ensure all tests are passing

**What are the relevant pivotal tracker stories?**
- [#160536352](https://www.pivotaltracker.com/story/show/160536352)

**Screenshots**

screen shot 2018-09-24 at 1 55 03 am
screen shot 2018-09-24 at 1 55 05 am
screen shot 2018-09-24 at 1 55 08 am

### CODE REVIEW

File: client/src/components/Admin/Meals/AddMealModal/Index.jsx [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+    element.click();
+  }
+
+  previewImage() {
+    const { current: element } = this.imageInput;
+    if (!element) return;
+
+    const image = element.files[0];
+    const status = validateAddMealImage(image);
+    if (status === true) {
```
Comment (monsieurkayode) [inline client/src/components/Admin/Meals/AddMealModal/Index.jsx]: No need for equality check here 
```js
if (status){}
```

Comment (abdulfataiaka) [reply client/src/components/Admin/Meals/AddMealModal/Index.jsx]: I needed to differentiate between `true` and `error string` that will be returned, as using your suggestion will result in both evaluating to true.

File: client/src/actions/admin/mealItemsAction.js [hunk 2]
```diff
@@ -35,3 +39,44 @@ export const fetchMealItems = () => dispatch => {
...
+
+export const showAddMealModal = show => dispatch => dispatch(
+  showAddMealModalAction(show)
+);
+
+export const addMealItem = formData => dispatch => {
+  dispatch(setAddMealLoading(true));
+
+  return axios.post(`${baseUrl}/admin/meal-items`, formData, {
+    headers: { 'content-type': 'application/json' }
```
[the remark this answers is not in the document]
Comment (abdulfataiaka) [reply client/src/actions/admin/mealItemsAction.js]: Did that as there might be issues if its not `json` by default

File: client/src/components/Admin/Meals/Index.jsx [hunk 3]
```diff
@@ -17,9 +20,7 @@ import { fetchMealItems } from '../../../actions/admin/mealItemsAction';
 export class Meals extends Component {
   constructor(props) {
     super(props);
-    this.state = {
-      addModalShow: false
-    };
+    this.state = {};
```
[the remark this answers is not in the document]
Comment (abdulfataiaka) [reply client/src/components/Admin/Meals/Index.jsx]: Mak sense.

File: client/src/components/Admin/Meals/Index.jsx [hunk 4]
```diff
@@ -95,12 +93,20 @@ export class Meals extends Component {
...
 
+const mapDispatchToProps = {
+  fetchMealItems,
+  showAddMealModal
+};
+
 Meals.propTypes = {
   fetchMealItems: PropTypes.func.isRequired,
   isLoading: PropTypes.bool.isRequired,
+  showAddMealModal: PropTypes.func,
```
[the remark this answers is not in the document]
Comment (abdulfataiaka) [reply client/src/components/Admin/Meals/Index.jsx]: Ok

File: client/src/components/Admin/Meals/AddMealModal/ImageView.jsx [hunk 5]
```diff
@@ -0,0 +1,41 @@
+/* eslint-disable jsx-a11y/img-redundant-alt */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const ImageView = (props) => {
+  const { openFileDialog, error, dataurl } = props;
+
+  if (dataurl == null) {
```
[the remark this answers is not in the document]
Comment (abdulfataiaka) [reply client/src/components/Admin/Meals/AddMealModal/ImageView.jsx]: Thats true

**Pull request merged**